### DISCUSSION TITLE
Does useForm destructuring have any effect on performance or bundle size?

### DISCUSSION THREAD

**Discussion by ifeltsweet** [react-hook-form/react-hook-form#6197] [Q&A] [answered]
Most examples destructure the return value of useForm:

```ts
const { register, handleSubmit } = useForm();
```

However, is there any advantage of doing the above as opposed to just doing the following?

```ts
const form = useForm();
```

react-hook-form is a little bit magic because it uses Proxy in certain cases, so I wanted to make sure I understand the impact of this correctly. Also, Webpack and other bundlers do various optimizations like tree shaking and whatnot – does this have any effect on it at all?

**Comment by bluebill1049** [comment, marked as answer]
there is no difference, only difference is with `formState`, it has a subscription modal with proxy.
